_posts/2022-03-16-solidity-0.8.13-release-announcement.md

```diff
@@ -0,0 +1,144 @@
+...
+
+## Important Bugs
+
+When ``abi.encodeCall`` was introduced in Solidity 0.8.11,
+hex literals (``0x1234``) and string literals (``"abcd"``) were not handled properly.
+Please read more about it in the [security alert](https://blog.soliditylang.org/2022/03/16/encodecall-bug/).
+
+## Notable New Features
+
+### Yul IR Pipeline Production Ready
+
+TODO
+
+talk about ``assembly ("memory-safe") { ... } `` or, if you must, ``/// @solidity memory-safe-assembly``
+
+Ask for feedback about gas costs and optimizer
+
+### Using For at File Level with Global Binding
+
+This release extends the ``using for`` directive considerably. It can now be used
+at file level, it is possible to use it with file-level functions and
+it has the option to make its effect global. We believe that in combination
+with the recently released user defined value types, this makes it possible
+to design proper reusable code libraries.
+
+In a future release, we plan to also add operators to user-defined types
+so that things like fixed point type libraries are relly easy to use
+and feel like compiler built-in types.
+
+If you use ``using {f, g, h} for Type;``, then the compiler already
+verifies that the given functions can receive a ``Type`` value as first
+argument.
+
+You can add the word ``global`` at the end to make the effect available
+everywhere. Since the idea is that you are defining a type and
+implementing the functions of the type, ``global`` can only be used
+for a user-defined type and only in the same file where the type is defined.
+We recommend that the ``using`` directive is always put directly after the definition
+of the type (forward references are valid in Solidity).
+
+As an example, let us consider a type that can only be incremented and decremented by one
+at a time. Because of this restriction, we can safely drop overflow checks
+since it would take more gas than is available in the universe to reach
+an overflow. We use the mechanism of an implementation file
+so that the names ``plusOne`` and ``minusOne`` do not pollute the namespace.
+
+File ``"restrictedNumber_impl.sol"``:
+```
+import RestrictedNumber from "./restrictedNumber.sol";
+function plusOne(RestrictedNumber x) pure returns (RestrictedNumber)
+{
+    unchecked {
+        return RestrictedNumber.wrap(RestrictedNumber.unwrap(x) + 1);
+    }
+}
+function minusOne(RestrictedNumber x) pure returns (RestrictedNumber)
+{
+    unchecked {
+        return RestrictedNumber.wrap(RestrictedNumber.unwrap(x) - 1);
+    }
+}
+```
+
+File ``"restrictedNumber.sol"``:
+```
+import "./RestrictedNumber_impl.sol" as RestrictedNumberImpl;
+type RestrictedNumber is int256;
+using {RestrictedNumberImpl.inc, RestrictedNumberImpl.dec} for RestrictedNumber global;
+```
+
+File ``"owned.sol"``:
+```
+import RestrictedNumber from "./restrictedNumber.sol";
+contract Owned {
+    RestrictedNumber public ownerCount;
+    mapping(address => bool) public isOwner;
```

Comment:
I'd recommend initializing these two variables to make the example somewhat complete.

Reply:
What do you mean by initialize?

Reply:
Have a constructor and do
```solidity
isOwner[msg.sender] = true;
ownerCount = ownerCount.plusOne();
```